python/lsst/dax/ppdb/sql/_base.py

```diff
@@ -0,0 +1,475 @@
+            raise IncompatibleVersionError(
+                f"Current code version {VERSION} is not compatible with database version {db_code_version}"
+            )
+
+    @classmethod
+    def read_schema(
+        cls, schema_file: str | None, schema_name: str | None, felis_schema: str | None, db_url: str
+    ) -> tuple[sqlalchemy.schema.MetaData, VersionTuple]:
+        """Read felis schema definitions for PPDB.
+
+        Parameters
+        ----------
+        schema_file : `str` or `None`
+            Name of YAML file with ``felis`` schema, if `None` then default
+            schema file is used.
+        schema_name : `str` or `None`
+            Database schema name, if `None` then default schema is used.
+        felis_schema : `str`, optional
+            Name of the schema in YAML file, if `None` then file has to contain
+            single schema.
+        db_url : `str`
+            Database URL.
+
+        Returns
+        -------
+        metadata : `sqlalchemy.schema.MetaData`
+            SQLAlchemy metadata instance containing information for all tables.
+        version : `lsst.dax.apdb.VersionTuple` or `None`
+            Schema version defined in schema or `None` if not defined.
+        """
+        if schema_file is None:
+            schema_file = os.path.expandvars(cls.default_felis_schema_file)
+
+        res = ResourcePath(schema_file)
+        schemas_list = list(yaml.load_all(res.read(), Loader=yaml.SafeLoader))
+        if not schemas_list:
+            raise ValueError(f"Schema file {schema_file!r} does not define any schema")
+        if felis_schema is not None:
+            schemas_list = [schema for schema in schemas_list if schema.get("name") == felis_schema]
+            if not schemas_list:
+                raise ValueError(f"Schema file {schema_file!r} does not define schema {felis_schema!r}")
+        elif len(schemas_list) > 1:
+            raise ValueError(f"Schema file {schema_file!r} defines multiple schemas")
+        schema_dict = schemas_list[0]
+
+        # Filter out unwanted tables. By default, all tables will be included.
+        filtered_tables = cls.filter_tables(schema_dict)
+
+        schema_dict["tables"] = filtered_tables
+        dm_schema: FelisSchema = felis.datamodel.Schema.model_validate(schema_dict)
+        schema = schema_model.Schema.from_felis(dm_schema)
+
+        # Replace schema name with a configured one, just in case it may be
+        # used by someone.
+        if schema_name:
+            schema.name = schema_name
+
+        # Create the PpdbReplicaChunk table and add it to the schema.
+        replica_chunk_table = cls.create_replica_chunk_table()
+        schema.tables.append(replica_chunk_table)
+
+        if schema.version is not None:
+            version = VersionTuple.fromString(schema.version.current)
+        else:
+            # Missing schema version is identical to 0.1.0
+            version = VersionTuple(0, 1, 0)
```

Comment:
I think at this point we can already require that `schema.version` must be defined.